eva-accession-clustering-automation/cluster_multiple_assemblies.py

```diff
@@ -0,0 +1,120 @@
+    """
+    preliminary_check(source, asm_vcf_prj_list, assembly_list)
+
+    if source.upper() == 'MONGO':
+        cluster_multiple_from_mongo(source, assembly_list, private_config_file, private_config_xml_file, profile,
+                                    output_directory, clustering_artifact, only_printing)
+
+    if source.upper() == 'VCF':
+        cluster_multiple_from_vcf(source, asm_vcf_prj_list, private_config_file, private_config_xml_file, profile,
+                                  output_directory, clustering_artifact, only_printing)
+
+
+def preliminary_check(source, asm_vcf_prj_list, assembly_list):
+    """
+    This checks must pass in order to run the script
+    """
+    check_valid_sources(source)
+    check_requirements(source, asm_vcf_prj_list, assembly_list)
+
+
+def cluster_multiple_from_mongo(source, assembly_list, private_config_file, private_config_xml_file, profile,
+                                output_directory, clustering_artifact, only_printing):
+    """
+    This method splits the list of assemblies and call the run_clustering method for each assembly
+    """
+    for assembly in assembly_list.split(','):
+        run_clustering(source, None, None, assembly, private_config_file, private_config_xml_file, profile,
+                       output_directory, clustering_artifact, only_printing, timestamp)
+
+
+def cluster_multiple_from_vcf(source, asm_vcf_prj_list, private_config_file, private_config_xml_file, profile,
+                              output_directory, clustering_artifact, only_printing):
+    """
+    The list will be of the form: asm1,GCA000000001.1,PRJEB1111#asm2,GCA000000002.2,PRJEB2222...
+    This method splits the triplets and then call the run_clustering method for each one
+    """
+    for triplet in asm_vcf_prj_list.split(','):
+        data = triplet.split('#')
+        run_clustering(source, data[1], data[2], data[0], private_config_file, private_config_xml_file, profile,
+                       output_directory, clustering_artifact, only_printing, timestamp)
+
+
+def check_requirements(source, asm_vcf_prj_list, assembly_list):
+    """
+    This method checks depending on the source, what list should have been provided.
+    For VCF it is expected to have a list of one or more assembly, vcf file, project separated by #
+    For Mongo it is expected to have a list of assemblies.
+    """
+    if source.upper() == 'VCF' and not asm_vcf_prj_list:
+        logger.error('If the source is VCF a list of assembly#vcf#project mus be provided using the parameter'
+                     '--asm-vcf-prj-list')
+        sys.exit(1)
+    if source.upper() == 'MONGO' and not assembly_list:
+        logger.error('If the source is MONGO a list of assembly accessions must be provided using the parameters '
+                     '-al or --assembly-list')
+        sys.exit(1)
+
+
+if __name__ == "__main__":
+    parser = argparse.ArgumentParser(description='Cluster multiple assemblies', add_help=False)
```

Comment:
I would suggest looking into [sub-commands](https://docs.python.org/3.8/library/argparse.html#sub-commands) which should allow you to do all the validation at the argument parsing stage

Reply:
I think this is a good idea to have different commands for running clustering from VCF or mongo but did not include that in this PR